### DISCUSSION TITLE
Automating provider inactivation with directory file loads?

### DISCUSSION THREAD

**Discussion by brentfriedman** [nextgenhealthcare/connect#4865] [General]
If this isn't the correct place for this question, please move/remove it.

When I prepare a provider directory to be loaded into Mirth (as a csv file), there is one column in which I enter the text 'active.'  This makes loading new providers easy.  However, I have been told that the only way to inactivate a provider is to manually process each entry.  Is there any method for bulk-inactivation, through the regular csv file load or some other process?

I have tried to locate this in the documentation, but so far I haven't found anything about the provider loading process.

Thanks for any assistance!

**Comment by jonbartels** [comment]
If Mirth Connect is used for this process it would still rely on channel code specifically for your use case. This may depend on how data is managed in your provider directory. For example a direct DB call can do bulk operations but an API call might not support bulk deletes.

In general - MC can be used to perform bulk updates, inserts, and deletes to a database from a CSV file as input. 

In general - If a downstream system can only support single updates then MC can accept a CSV in bulk and dole out the individual updates.

If you can share it safely and legally - posting the channel that you are using may help get some specific advice. There is also an active Slack community that is good for real-time discussion as well as various consulting agencies if you need more hands on help.

**Reply by brentfriedman** [reply]
I will see if the data can be shared. It isn't secret, but would identify an organization.  Thanks!

**Reply by tonygermano** [reply]
If @jonbartels didn't make it clear enough. The provider loading process you are asking about is not part of mirth. It would be something that was custom built for your instance.
